src/plugins/intel_gpu/src/graph/include/reshape_inst.h

```diff
@@ -73,7 +73,9 @@ struct typed_program_node<reshape> : public typed_program_node_base<reshape> {
     }
 
     bool has_padding() const {
-        return (this->get_output_layout().data_padding || input().get_output_layout(false).data_padding || input().get_output_layout(false).has_dynamic_pad());
+        return (this->get_output_layout().data_padding
+                || input().get_output_layout(false).data_padding
+                || input().get_output_layout(false).data_padding.is_dynamic_pad());
```

Comment:
Remove `input().get_output_layout(false).data_padding.is_dynamic_pad()` if you add mask check to operator bool